internal/server/ranking/ranking.go

```diff
@@ -105,53 +108,59 @@ type CohortByRank []*pb.SourceSeries
 // score, otherwise can also match to wildcard options (indicated by *).
 //
 // If no entry is found, a BaseRank is assigned to the source series.
+func GetScoreRk(importName string, rk RankKey) int {
+    importNameStatsRanking, ok := StatsRanking[importName]
+    if ! ok {
+        return BaseRank
+    }
+    scoreOfMostExactMatch := BaseRank
+    mostMatches := -1
+    for k, score := range importNameStatsRanking {
+        matches := 0
+        isMatch := true
+        if k.MeasurementMethod != "*" {
+            if k.MeasurementMethod == rk.MeasurementMethod {
+                matches++
+            } else {
+                isMatch = false
+            }
+        }
+
+        if k.ObservationPeriod != "*" {
+            if k.ObservationPeriod == rk.ObservationPeriod {
+                matches++
+            } else {
+                isMatch = false
+            }
+        }
+
+        isMoreExactMatch := matches > mostMatches
+        isSameExactnessMatchWithBetterScore := score < scoreOfMostExactMatch && matches == mostMatches
+        if isMatch && (isMoreExactMatch || isSameExactnessMatchWithBetterScore) {
+            scoreOfMostExactMatch = score
+            mostMatches = matches
+        }
```

Comment:
If I understand correctly, line 137:142 can be written as before:

if !isMatch {
  continue
}
if matches < mostMatches {
  continue
}
if matches == mostMatches && score > rankScore {
  continue
}
bestScore = score
mostMatches = matches

Reply:
Basically you want to use concise conditions and linear logic instead of complex conditions.

Also, I am not sure if we have the case of "matches == mostMatches && score < rankScore"

Reply:
> Also, I am not sure if we have the case of "matches == mostMatches && score < rankScore"

Line 138: `isSameExactnessMatchWithBetterScore := score < scoreOfMostExactMatch && matches == mostMatches` handles this.